src/functional/ffront/foast_passes/type_deduction.py

```diff
@@ -58,6 +59,93 @@ def check_type_refinement(node: foast.Expr, new: foast.DataType) -> None:
             )
 
 
+@dataclass
+class TypeInfo:
+
+    type: foast.SymbolType  # noqa: A003
+
+    @property
+    def is_complete(self) -> bool:
+        return is_complete_symbol_type(self.type)
```

Comment:
Not related to this PR, but I wonder if it makes sense to define `is_complete_symbol_type()` as a `TypeGuard[foast.SymbolType]` if it returns `False` for `foast.DeferredSymbolType`, which is a subclass of `foast.SymbolType`

Reply:
I was wondering that too when I followed your suggestion to do so.

Reply:
Yes, I remember the conversation but either I hadn't realized  that `DeferredSymbolType` was a subclass of `SymbolType` or I was just wrong.